### PR TITLE
Move core/routemgmt from openwhisk repo to openwhisk-apigateway

### PR DESCRIPTION
**Pull request by ServoKvd**

### CODE REVIEW

**Comment by ServoKvd** [comment]
To be merged before https://github.com/apache/incubator-openwhisk/pull/3056

File: routemgmt/common/apigw-utils.js [hunk 1]
```diff
@@ -0,0 +1,896 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Route management action common API GW utilities
+ */
+var request = require('request');
```
Comment (mhamann) [inline routemgmt/common/apigw-utils.js]: Use `request-promise` instead

File: routemgmt/common/apigw-utils.js [hunk 2]
```diff
@@ -0,0 +1,896 @@
...
+ * @param    gwAuth    Required. The user bearer token used to access the API Gateway REST endpoints
+ * @param spaceGuid    Required. User's space guid.  APIs are stored under this context
+ * @param swaggerApi   Required. The API swagger object to send to the API gateway
+ * @param apiId        Required. API id. When specified, the API exists and will be updated; otherwise the API is created anew
+ * @return A promise for an object describing the result with fields error and response
+ */
+function addApiToGateway(gwInfo, spaceGuid, swaggerApi, apiId) {
+  var requestFcn = request.post;
+
+  console.log('addApiToGateway: ');
```
Comment (mhamann) [inline routemgmt/common/apigw-utils.js]: Remove console.log statements. If logging is required, use `debug` or `bunyan` etc.

File: routemgmt/common/apigw-utils.js [hunk 3]
```diff
@@ -0,0 +1,896 @@
...
+    options.url = gwInfo.gwUrl + '/' + encodeURIComponent(spaceGuid) + '/apis/' + encodeURIComponent(apiId);
+    requestFcn = request.put;
+  }
+
+  console.log('addApiToGateway: request: '+JSON.stringify(options, " ", 2));
+  }
+  catch (e) {
+    console.error('addApiToGateway exception: '+e);
+  }
+  return new Promise(function(resolve, reject) {
```
Comment (mhamann) [inline routemgmt/common/apigw-utils.js]: Don't make new promises. Use promisification or libraries that export promises.

Comment (mhamann) [reply routemgmt/common/apigw-utils.js]: Also, prefer async/await if using Node > 8

**Comment by csantanapr** [comment]
Sorry @mhamann this PR is invalid the actions will be move to CLI repo instead

**Pull request closed without merging**